src/lib.rs

```diff
@@ -144,6 +144,17 @@ impl Ulid {
     pub fn is_nil(&self) -> bool {
         self.0 == 0u128
     }
+
+    /// increment the random number, make sure that the ts millis stays the same
+    fn increment(&self) -> Option<Ulid> {
+        let prev_ts_ms = self.timestamp_ms();
+        let next = Ulid(self.0 + 1);
```

Comment:
Addition is not safe here as `self.0` could be `u128::max_value()`. Please mask out the random bits and compare that to the max random bits value.

Reply:
I changed this it [wrapping_add](https://doc.rust-lang.org/std/primitive.u32.html#method.wrapping_add) if you think that it would be clearer to use the mask and check for max value i can do that instead. `wrapping_add` is what I originally intended.

Reply:
Ulid only has 80 bits of randomness, so you may not want it to overflow to the timestamp bits.